### PR TITLE
All Projects: Handle LICENSE.txt and SECURITY.md automatically.

### LINKED ISSUE #19801: All Projects: Ensure there is a LICENSE file in the project root
While we mention the license in the `composer.json` file of each project that everything is GPLv2+, since we now have mirror repos, we really should include a LICENSE file or some other form of including the license in every project.

Discussion to confirm GPL2+ is the right license for all the things: p2y3YZ-4v3-p2

### LINKED ISSUE #19802: CLI Generate: Add LICENSE file to templates
Related to https://github.com/Automattic/jetpack/issues/19801, let's add the GPL v2 as a LICENSE file in the skeleton used for all projects.

### PR DESCRIPTION
**Pull request by kraftbj**
- Add full GPL to the CLI generate skeleton Closes #19802
- Add GPL info to all plugin projects
- Add auto-licensing to build step Closes #19801
- Auto-copy security notice file

#### Changes proposed in this Pull Request:

* Adds automation around LICENSE and SECURITY files

#### Jetpack product discussion

p2y3YZ-4v3-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* On the build action, ensure no errors for the LICENSE copying.
* Verify the artifact includes LICENSE.txt and SECURITY.md in every project.

### CODE REVIEW

**Comment by kraftbj** [comment]
While the FSF says the preface should be in every source file with the full license included once, we haven't done that. I compromised in this PR by putting the preface at the top of the "entry" files for each plugin and the js-packages. As actions and the packages don't have a good entry file, I omitted it.

Additionally, I'm copying from a file within our tree to 1. not rely on a third party service and 2. ensure that we only license code with licenses approved by the Automattlock team. While we have other licenses we've used at a8c, we have a strong philosophical support and history with the GPL2+.

At this time, we're only adding GPL2+ as an approved license for our monorepo code, just so we ensure we have a conversation before licensing something else.

**Review by jeherve — DISMISSED** [review]
This looks good to me. 🚢 ⚖️

File: .github/files/build-all-projects.sh [hunk 1]
```diff
@@ -141,6 +141,22 @@ for SLUG in "${SLUGS[@]}"; do
...
+	if [[ "$LICENSE" ]]; then
+		echo "License: $LICENSE"
+		if cp "$BASE/.github/licenses/$LICENSE.txt" "$BUILD_DIR/LICENSE.txt"; then
+			echo "License file copied."
+		else
+			echo "License value not approved."
+			EXIT=1
+			continue
+		fi
+	fi
```
Comment (anomiex) [inline .github/files/build-all-projects.sh:155]:
```suggestion
	else
		echo "No license declared."
		# TODO: Make this an error?
	fi
```

**Review by anomiex — COMMENTED** [review]
> While the FSF says the preface should be in every source file with the full license included once, we haven't done that.

Personally, I'm fine with not having 15 lines of license crap at the top of every file. Especially when, once you add it, you can never remove it without permission from everyone who contributed to the file.

If we decide we want license indication in each file, IMO `@license` should be enough.

File: .github/files/build-all-projects.sh [hunk 2]
```diff
@@ -141,6 +141,22 @@ for SLUG in "${SLUGS[@]}"; do
...
 	fi
 
+	# Copy license.
+	LICENSE=$(jq -r '.license // ""' composer.json)
+	if [[ "$LICENSE" ]]; then
+		echo "License: $LICENSE"
+		if cp "$BASE/.github/licenses/$LICENSE.txt" "$BUILD_DIR/LICENSE.txt"; then
+			echo "License file copied."
+		else
+			echo "License value not approved."
```
Comment (anomiex) [inline .github/files/build-all-projects.sh:151]: Flag the error for GH.
```suggestion
			echo "::error file=projects/$SLUG/composer.json::License value not approved."
```

File: .github/files/build-all-projects.sh [hunk 3]
```diff
@@ -141,6 +141,22 @@ for SLUG in "${SLUGS[@]}"; do
 		cp -r "$BASE/.github/files/gh-autotagger/." "$BUILD_DIR/.github/."
 	fi
 
+	# Copy license.
+	LICENSE=$(jq -r '.license // ""' composer.json)
+	if [[ "$LICENSE" ]]; then
```
Comment (anomiex) [inline .github/files/build-all-projects.sh:146]: I like to be a little more explicit in these tests. I think bash is smart enough to not break even if someone sets the license to something that looks like an operator, but it's nice to not have to worry about it.
```suggestion
	if [[ -n "$LICENSE" ]]; then
```

**Review by anomiex — APPROVED** [review]
Seems to be working.

Personally I'd leave out adding the GPL header cruft to the files you're adding it to for the reasons I stated earlier, but I won't withhold approval based on that.

**Pull request merged**